checkfiles/checkfiles.py

```diff
@@ -183,6 +184,180 @@ def check_format(encValData, job, path):
+                'fastq file contains mixed read numbers ' + \
+                '{}.'.format(', '.join(sorted(list(read_numbers_set))))
+
+        # read_length
+        read_lengths_list = []
+        for k in sorted(read_lengths_dictionary.keys()):
+            read_lengths_list.append((k, read_lengths_dictionary[k]))
+
+        if 'read_length' in item and item['read_length'] > 2:
+            process_read_lengths(read_lengths_dictionary,
+                                 read_lengths_list,
+                                 item['read_length'],
+                                 read_count,
+                                 0.95,
+                                 errors)
+        else:
+            errors['read_length'] = 'no specified read length in the uploaded fastq file, ' + \
+                                    'while read length(s) found in the file were {}.'.format(
+                                    ', '.join(read_lengths_list))
+
+        # signatures
+        uniqueness_flag = True
+        signatures_for_comparison = set()
+        is_UMI = False
+        if 'flowcell_details' in item and len(item['flowcell_details']) > 0:
+            for entry in item['flowcell_details']:
+                if 'barcode' in entry and entry['barcode'] == 'UMI':
+                    is_UMI = True
+                    break
+        if is_UMI is True:
+            for entry in signatures_no_barcode_set:
+                signatures_for_comparison.add(entry + 'UMI:')
+        else:
+            if len(signatures_set) > 100:
+                for entry in signatures_no_barcode_set:
+                    signatures_for_comparison.add(entry + 'mixed:')
+            else:
+                signatures_for_comparison = signatures_set
+
+        conflicts = []
+        for unique_signature in signatures_for_comparison:
+            query = '/' + unique_signature + '?format=json'
+            r = session.get(urljoin(url, query))
+            response = r.json()
+            if response is not None and 'File' in response['@type']:
+                uniqueness_flag = False
+                conflicts.append(
+                    'specified unique identifier {} '.format(unique_signature) +
+                    'is conflicting with identifier of reads from ' +
+                    'file {}.'.format(response['accession']))
+        if uniqueness_flag is True:
+            result['fastq_signature'] = sorted(list(signatures_for_comparison))
+        else:
+            errors['not_unique_flowcell_details'] = conflicts
+
+    except IOError:
+        errors['file_open_error'] = 'OS could not open the file ' + \
+                                    unzipped_fastq_path
+
+
+def process_read_lengths(read_lengths_dict,
+                         lengths_list,
+                         submitted_read_length,
+                         read_count,
+                         threshold_percentage,
+                         errors_to_report):
+    expected_read_lengths = [
```

Comment:
the current decision is to check whether 95% of read lengths in the file are close to reported read length in the metadata. Since we allow deviation of +-2bp in other places (length standards) I used this cutoff here as well - if it will not work we can narrow/widen acordingly